neqo-transport/src/lib.rs

```diff
@@ -209,13 +209,17 @@ impl ::std::fmt::Display for Error {
 
 pub type AppError = u64;
 
+#[deprecated(note = "use `CloseReason` instead")]
+pub type ConnectionError = CloseReason;
```

Comment:
```suggestion
pub type ConnectionCloseReason = CloseReason;
```
?

Reply:
Ah I guess we can't rename this one.

Reply:
Users that upgrade to this patch might still be using the old `ConnectionError`. In order to give them a compiler warning, hinting to them that they should use `CloseReason` instead, these two lines are added.

The type alias (i.e. `pub type ConnectionError = CloseReason;`) will make sure their code still compiles, as `ConnectionError` is now simply an alias for the new `CloseReason`. The `deprecated` attribute will give them the warning that they are using a deprecated type (i.e. `ConnectionError`) and that they should switch to `CloseReason`.

Does that make sense?

Reply:
Here is a sample compiler warning that a user would get:

```
warning: use of deprecated type alias `ConnectionError`: use `CloseReason` instead
    |
119 |     error: ConnectionError,
    |            ^^^^^^^^^^^^^^^